nomad/rpc.go

```diff
@@ -140,6 +147,34 @@ func (ctx *RPCContext) ValidateCertificateForName(name string) error {
 	return fmt.Errorf("invalid certificate, %s not in %s", name, strings.Join(validNames, ","))
 }
 
+func (ctx *RPCContext) IsStatic() bool {
+	return ctx == nil
+}
+
+func (ctx *RPCContext) GetRemoteIP() (net.IP, error) {
+	if ctx == nil {
+		return nil, nil
```

Comment:
Is this an expected outcome? Would it make sense to return an error instead of a potentially dangerous `nil, nil`?

Reply:
Yes, we'll see this a lot! The `RPCContext` is nil whenever we have "static" RPCs, which are the RPCs sent as normal (albeit more expensive) method calls from components internal to the process. So for example, the deploymentwatcher only makes RPCs internal to the leader and doesn't go over wire (except for MRD), so there's no "remote IP" to get for that case.

Reply:
Ah got it, thanks!